### PR TITLE
Add higher order Container component to support registry items

### LINKED ISSUE #512: Support `registry` items using a Container
**Enhancement** 

It should be possible to support registry items using a container, currently this is not possible with the `Container` as a mixin (as it needs to extend a component constructor).

If the `Container` was converted to be a "higher order component" we could support both component constructor and registry items.

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Add support for registry items with the `Container` by converting the from a mixin to a higher order component.

Resolves #512

### CODE REVIEW

File: src/Container.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { WidgetBase } from './WidgetBase';
+import { Constructor, DNode, WidgetBaseInterface, RegistryLabel } from './interfaces';
+import { w } from './d';
+import { defaultMappers, BaseInjector } from './Injector';
+
+export function Container<W extends WidgetBaseInterface>(
+	component: Constructor<W> | RegistryLabel,
+	name: string,
```
Comment (matt-gadd) [inline src/Container.ts]: This should/could probably be a `RegistryLabel`

File: src/Container.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	{ getProperties = defaultMappers.getProperties, getChildren = defaultMappers.getChildren }: any = defaultMappers
```
Comment (matt-gadd) [inline src/Container.ts]: the formatting of this is difficult to read, can you multi line it

Comment (matt-gadd) [inline src/Container.ts]: can the `any` be typed?

File: src/Container.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+): Constructor<WidgetBase<W['properties']>> {
+
+	return class extends WidgetBase<any> {
+		protected render(): DNode {
+			const { properties, children } = this;
+
+			return w<BaseInjector<any>>(name, {
+				bind: this,
+				render: () => { return w(component, properties, children); },
```
Comment (matt-gadd) [inline src/Container.ts]: if this explicitly returns you can drop the braces and `return` statement

File: src/Container.ts (hunk 2 above)
Comment (agubler) [inline src/Container.ts]: It can. done.

File: src/Container.ts (hunk 3 above)
Comment (agubler) [inline src/Container.ts]: It can, tend not to by default. Updated all the same.

**Pull request merged**